StudyfiedBackend/Controllers/Quiz/QuizHelper.cs

```diff
@@ -100,14 +101,30 @@ public static bool isValidQuiz(Quiz quiz, int numberOfQuestion)
 
             bool isValidQuiz = true;
 
-            bool isValidNumberOfQuestions = quiz.questionAnswerPairs.Keys
-                .Select(question => question.question != null && question.question != "")
-                .Count() == numberOfQuestion;
+            int numOfQuizQuestions = 0;
```

Comment:
this method is for validating a quiz object
a validation method should not have any mutations ( you are mutating the quiz by removing questions)
you simply have to assert that the number of questions you asked for is correct, while checking that each question is valid ( non null non empty question + non null non empty answers for given question)